src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -112,23 +115,27 @@ object Task {
     //scalastyle:off cyclomatic.complexity method.length
     override def update(update: TaskStateOp): TaskStateChange = update match {
       // case 1: now running
-      case TaskStateOp.MesosUpdate(_, MarathonTaskStatus.Running(mesosStatus), now) if !hasStartedRunning =>
+      case TaskStateOp.MesosUpdate(_, MarathonTaskStatus.Running, mesosStatus, now) if !hasStartedRunning =>
```

Comment:
with a dedicated status, we no longer need to infer the state by evaluating `hasStartedRunning`.

This can be something like
`case TaskStateOp.MesosUpdate(_, MarathonTaskStatus.Running, mesosStatus, now) if this.isStaging =>`

Not sure whether `Starting` would be possible as well